src/Mock.spec.ts

```diff
@@ -255,4 +255,37 @@ describe('jest-mock-extended', () => {
             expect(mockObj.deepProp.getAnotherString('abc')).toBe('this string');
         });
     });
+
+    describe('Promise', () => {
+        test('Can return as Promise.resolve', async () => {
+            const mockObj = mock<MockInt>();
+            mockObj.id = 17;
+            const promiseMockObj = Promise.resolve(mockObj);
+            await expect(promiseMockObj).resolves.toBeDefined();
+            await expect(promiseMockObj).resolves.toMatchObject({id :17});
+        });
+        test('Can return as Promise.reject', async () => {
+            const error = new Error('17');
```

Comment:
should this not be a mock here?

Reply:
Oh, Yes it is true. I will fix it today

Reply:
I had problem with expect(mock).toEqual(mock), because in a method toEqual call iterator for the mock and own mock return not correct data for an iterator. You can see more information in https://medium.com/front-end-weekly/thank-u-symbol-iterator-next-aef9f09ff78. I used expect(mock.property).toEqual(mock.property)

it is not critical for me but in future it need fix.

Reply:
Cool. I can take a look at this.